backend/src/main/java/com/aadeshandreas/ailearning/ai_learning_companion/service/DocumentService.java

```diff
@@ -83,6 +83,26 @@ public void uploadDocument(MultipartFile pdfFile) throws IOException {
             Files.deleteIfExists(tempFile);
         }
     }
+    /**
+     * Accepts raw text input and stores it in the current session's repository
+     * clears previously stored summaries/flashcards
+     * @param text raw text tha's provided by user
+     */
+    public void uploadText(String text){
+        if(text == null || text.trim().isEmpty()){
+            throw new IllegalArgumentException("Text cannot be empty");
+        }
+
+        documentRepository.setDocumentText(text.trim());
+        clearCache();
+    }
```

Comment:
```suggestion
    public void uploadText(String text){
        documentRepository.setDocumentText(text.trim());
        clearCache();
    }
```

The DTO handles the null and empty check so we dont have to check it here again.